chia/data_layer/data_store.py

```diff
@@ -484,6 +484,21 @@ async def get_first_generation(self, node_hash: bytes32, store_id: bytes32) -> O
 
             return int(row[0])
 
+    async def get_existing_hashes(self, node_hashes: list[bytes32], store_id: bytes32) -> set[bytes32]:
+        result: set[bytes32] = set()
+        batch_size = min(500, SQLITE_MAX_VARIABLE_NUMBER - 10)
+
+        for i in range(0, len(node_hashes), batch_size):
+            chunk = node_hashes[i : i + batch_size]
+            placeholders = ",".join(["?"] * len(chunk))
+            query = f"SELECT hash FROM nodes WHERE store_id = ? AND hash IN ({placeholders})"
+            async with self.db_wrapper.reader() as reader:
+                cursor = await reader.execute(query, (store_id, *chunk))
+                async for row in cursor:
+                    result.add(row[0])
```

Comment:
Making sure the cursor gets closed.

```suggestion
                async with reader.execute(query, (store_id, *chunk)) as cursor:
	                async for row in cursor:
	                    result.add(row[0])
```

Or maybe also avoid the repeated async yielding since we have a limited size here anyways.  I don't know how much of a difference this would actually make, if even noticeable.

```suggestion
                async with reader.execute(query, (store_id, *chunk)) as cursor:
	                rows = await cursor.fetchall()
                        result.update(row[0] for row in rows)
```

And I think the row can be used as a dict?

```suggestion
                async with reader.execute(query, (store_id, *chunk)) as cursor:
	                rows = await cursor.fetchall()
                        result.update(row["hash"] for row in rows)
```